fedimint-client/src/lib.rs

```diff
@@ -743,3 +844,36 @@ pub fn client_decoders<'a>(
     }
     Ok(ModuleDecoderRegistry::from_iter(modules))
 }
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct OperationLogEntry {
+    operation_type: String,
+    meta: serde_json::Value,
+}
+
+impl Encodable for OperationLogEntry {
+    fn consensus_encode<W: Write>(&self, writer: &mut W) -> Result<usize, Error> {
+        let mut len = 0;
+        len += self.operation_type.consensus_encode(writer)?;
+        len += serde_json::to_string(&self.meta)
+            .expect("JSON serialization should not fail")
+            .consensus_encode(writer)?;
+        Ok(len)
+    }
+}
+
+impl Decodable for OperationLogEntry {
+    fn consensus_decode<R: Read>(
+        r: &mut R,
+        modules: &ModuleDecoderRegistry,
+    ) -> Result<Self, DecodeError> {
+        let operation_type = String::consensus_decode(r, modules)?;
+        let meta_str = String::consensus_decode(r, modules)?;
+        let meta = serde_json::from_str(&meta_str).map_err(DecodeError::from_err)?;
```

Comment:
Sneaking in `SerdeEncodable` here like that, just after you removed it? :P

Reply:
Logging json as a free form metadata is not unusual, but I think it's funny anyway. :D